### DISCUSSION TITLE
how to preserve jsx

### DISCUSSION THREAD

**Discussion by stefanprobst** [swc-project/swc#2747] [Q&A] [answered]
hi, is it currently possible to preserve jsx (i.e. not transform it)?

**Comment by kdy1** [comment, marked as answer]
No, at least currently if you are using javascript api

**Comment by maor-benami** [comment]
Can we please get a jsx:preserve config option?

**Comment by yourWaifu** [comment]
Can I get an update to this? I would like to use SWC, but this is the one thing holding me back and keeping me with Babel.

**Reply by XantreDev** [reply]
It's really sad. In my case i want to test my plugin, but my snapshots uses jsx syntax. Will compile snapshots too(

**Comment by kdy1** [comment]
https://github.com/swc-project/swc/issues/9929

**Comment by shanshrew** [comment]
@kdy1 Legendary. can we get this released? really need this actually been waiting on it

**Reply by kdy1** [reply]
No, as this should be released as v2

**Reply by shanshrew** [reply]
I know this probably isn't the right thread for such discussion. But can you release an early build for 2.0.0? or is there a place to see the 2.0.0 roadmap somewhere?

We have a swc plugin that needs to run in our pipeline independently of everything downstream (For reasons outside of the scope of this post).

This prevents us from using swc and instead forces us to use babel, because if we use SWC JSX is removed and anything downstream that needs to analyze JSX can't anymore.
